components/CheckoutWizard/CheckoutWizard.tsx

```diff
@@ -33,12 +33,24 @@ export const CheckoutWizard: FunctionComponent<Props> = ({ assetName, children }
   const [currentStep, setCurrentStep] = useState(0)
   const router = useRouter()
   const intl = useIntl()
-  const { removeItem } = useCheckoutData()
+  const { checkoutData, removeItem } = useCheckoutData()
 
   useEffect(() => {
     const stepByPathname = steps.findIndex((step) => router.pathname.includes(step))
+
+    if (stepByPathname > 0 && !(checkoutData as any)[steps[stepByPathname - 1]]) {
```

Comment:
Are we not able to type the checkoutdata?

Reply:
we are, but `redirect` doesn't exist on `checkoutData` which typescript knows because of how `steps` is typed, so it gets mad when we use `steps` to index `checkoutData`